### PR TITLE
Throttle how many rooms a user can join at once

### PR DESCRIPTION
**Pull request by mia-pi-git**
Discussed in Staff room.

### CODE REVIEW

**Comment by Slayer95** [comment]
This is not how you fail joining. You need to send ``|noinit|``

File: server/chat-commands/moderation.ts [hunk 1]
```diff
@@ -412,11 +412,17 @@ export const commands: ChatCommands = {
 	j: 'join',
 	async join(target, room, user, connection) {
 		if (!target) return this.parse('/help join');
+		const curRooms = [...Rooms.rooms.values()].filter(r => user.id in r.users);
+		target = target.trim();
```
Comment (Zarel) [inline server/chat-commands/moderation.ts:416]: I'd put this above the check for empty target.

Comment (Zarel) [inline server/chat-commands/moderation.ts:416]:
```suggestion
		target = target.trim();
 		if (!target) return this.parse('/help join');
```

File: server/chat-commands/moderation.ts [hunk 2]
```diff
@@ -412,11 +412,17 @@ export const commands: ChatCommands = {
...
 		if (target.startsWith('http://')) target = target.slice(7);
 		if (target.startsWith('https://')) target = target.slice(8);
 		if (target.startsWith(`${Config.routes.client}/`)) target = target.slice(Config.routes.client.length + 1);
 		if (target.startsWith(`${Config.routes.replays}/`)) target = `battle-${target.slice(Config.routes.replays.length + 1)}`;
 		if (target.startsWith('psim.us/')) target = target.slice(8);
+		// isn't in tryJoinRoom so you can still join your own battles / gameRooms etc
+		if (!user.can('altsself') && !target.startsWith('view-') && curRooms.length >= 50) {
+			return connection.sendTo(target as RoomID, `|noinit||You can only join 50 rooms at a time.`);
+		}
```
Comment (Zarel) [inline server/chat-commands/moderation.ts:425]:
```suggestion
		// isn't in tryJoinRoom so you can still join your own battles / gameRooms etc
		const numRooms = [...Rooms.rooms.values()].filter(r => user.id in r.users).length;
		if (!user.can('altsself') && !target.startsWith('view-') && numRooms >= 50) {
			return connection.sendTo(target as RoomID, `|noinit||You can only join 50 rooms at a time.`);
		}
```

File: server/chat-commands/moderation.ts [hunk 3]
```diff
@@ -411,12 +411,18 @@ export const commands: ChatCommands = {
 	joim: 'join',
 	j: 'join',
 	async join(target, room, user, connection) {
-		if (!target) return this.parse('/help join');
+		target = target.trim();
+ 		if (!target) return this.parse('/help join');
```
Comment (Zarel) [inline server/chat-commands/moderation.ts:415]:
```suggestion
		if (!target) return this.parse('/help join');
```

**Comment by Zarel** [comment]
Looks good! Thanks!

**Pull request merged**